core/trino-main/src/main/java/io/trino/sql/analyzer/CanonicalizationAware.java

```diff
@@ -91,6 +100,15 @@ public static Boolean canonicalizationAwareComparison(Node left, Node right)
         return null;
     }
 
+    public static Boolean canonicalizationAwareIgnoreCaseComparison(Node left, Node right)
```

Comment:
If we are ignoring case comparison - Say if we have a `ORDER BY x, "X"` - How would it match ?